### PR TITLE
mv `ParsedProof` to dedicated file with tests

### PR DESCRIPTION
**Pull request by tcoratger**
Verifier file is already quite big so that `ParsedProof` is moved to a dedicated file with unit tests

### CODE REVIEW

File: src/whir/parsed_proof.rs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                    .iter()
+                    .map(|answers| {
+                        CoefficientList::new(answers.clone()).evaluate(&round.folding_randomness)
+                    })
+                    .collect()
+            })
+            .collect();
+
+        // Add final round if needed
+        if !self.final_randomness_answers.is_empty() {
```
Comment (tcoratger) [inline src/whir/parsed_proof.rs:53]: @WizardOfMenlo Here please double check, but it seems that if `final_randomness_answers` we don't need to push anything. Before even if `final_randomness_answers` was empty, we had at the end `[[]]`

Comment (WizardOfMenlo) [reply src/whir/parsed_proof.rs:53]: Why can final_randomness_answers be empty? In theory these are the last Merkle answers which should always be there?

Comment (tcoratger) [reply src/whir/parsed_proof.rs:53]: Yes, in theory, in the protocol, I imagine this should never happen, but I came across this edge case in the unit tests, so I thought it was better to do it this way.

But let me know if you feel this is useless, and I can remove this check.

Comment (WizardOfMenlo) [reply src/whir/parsed_proof.rs:53]: I think we should just disallow this in the parsing, the protocol doesn't make much sense if the final answers (or any really are empty), and the verifier should reject.

Comment (tcoratger) [reply src/whir/parsed_proof.rs]: yeah that makes sense, for now, just reverted this to keep as it was before for this check

**Review by WizardOfMenlo — APPROVED** [review]
